src/main/java/com/sele3/page/leapfrog/LeapFrogPage.java

```diff
@@ -0,0 +1,174 @@
+package com.sele3.page.leapfrog;
+
+import com.codeborne.selenide.WebDriverRunner;
+import com.sele3.data.leapfrog.GameData;
+import com.sele3.utils.Assertion;
+import com.sele3.utils.Common;
+import com.sele3.utils.ExcelUtil;
+import io.qameta.allure.Allure;
+import io.qameta.allure.Step;
+import lombok.SneakyThrows;
+import org.apache.poi.ss.usermodel.*;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.codeborne.selenide.Selenide.open;
+
+public class LeapFrogPage {
+
+    private final String BASE_URL = "https://store.leapfrog.com/en-us/apps/c?p=%d&platforms=197&product_list_dir=asc&product_list_order=name";
+
+    public static List<GameData> parseGameDataFromHtml(String html) {
+        List<GameData> gameList = new ArrayList<>();
+        Document doc = Jsoup.parse(html);
+        Elements cards = doc.select("div.resultList div.catalog-product");
+        for (Element card : cards) {
+            String title = Common.normalize(card.select(".heading a").text());          // (French) Une belle journée pour rugir
+            String age = Common.normalize(card.select(".ageDisplay").text());           // Ages 4-6 years
+            String price = Common.normalize(card.select(".single.price").text().split(":")[1]);       // $7.50
+
+            gameList.add(new GameData(title.trim(), age.trim(), price.trim()));
+        }
+        return gameList;
+    }
+
+    @SneakyThrows
+    @Step("Read expected data from Excel")
+    public List<GameData> readExpectedData(String excelPath) {
```

Comment:
this method should be readFromExcelFile and the argument could be File type

Reply:
update